enzyme/Enzyme/Utils.cpp

```diff
@@ -2038,3 +2183,110 @@ llvm::FastMathFlags getFast() {
     f.set();
   return f;
 }
+
+void addValueToCache(llvm::Value *arg, bool cache_arg, llvm::Type *ty,
+                     llvm::SmallVector<llvm::Value *, 2> &cacheValues,
+                     llvm::IRBuilder<> &BuilderZ, llvm::Twine name) {
+   if (!cache_arg) return;
+  auto PT = cast<PointerType>(arg->getType());
+  if (PT->getElementType() != ty)
+    arg = BuilderZ.CreatePointerCast(arg, PointerType::get(ty, PT->getAddressSpace()), "pcld." + name);
+#if LLVM_VERSION_MAJOR > 7
+  arg = BuilderZ.CreateLoad(ty, arg, "avld." + name);
+#else
+  arg = BuilderZ.CreateLoad(arg, "avld." + name);
+#endif
+  cacheValues.push_back(arg);
+}
+
+void extractValueFromCache(llvm::Value *arg, bool cache_arg,
+                           llvm::Value *true_arg, llvm::Type *ty,
+                           llvm::Value *cacheval, unsigned cacheidx,
+                           DerivativeMode Mode,
+                           llvm::IRBuilder<> &allocationBuilder,
+                           llvm::IRBuilder<> &Builder2) {
```

Comment:
nit: In .cpp files we usually use namespace llvm, so we can avoid the llvm namespace prefix here